src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
>I would go as far as asserting here?

Does it make a difference here? Using `enforce` allows me to unit test it using `testThrown()`, but I guess I won't be able to do that if I use `assert` instead.